src/Draco.Compiler/Internal/Declarations/MergedModuleDeclaration.cs

```diff
@@ -12,20 +12,28 @@ internal sealed class MergedModuleDeclaration : Declaration
     public override ImmutableArray<Declaration> Children => this.children ??= this.BuildChildren();
     private ImmutableArray<Declaration>? children;
 
+    public string FullName { get; }
+
     private readonly ImmutableArray<SingleModuleDeclaration> declarations;
 
-    public MergedModuleDeclaration(ImmutableArray<SingleModuleDeclaration> declarations)
-        : base(declarations.FirstOrDefault()?.Name ?? string.Empty)
+    public MergedModuleDeclaration(string name, string fullName, ImmutableArray<SingleModuleDeclaration> declarations)
+        : base(name)
     {
         this.declarations = declarations;
+        this.FullName = fullName;
     }
 
     private ImmutableArray<Declaration> BuildChildren()
     {
+        var parentNesting = this.FullName.Count(x => x == '.');
```

Comment:
This is the reason I'd suggest using `SplitPath` and not just raw strings. This is currently easy to bamboozle